tests/contrib/django/test_django_appsec.py

```diff
@@ -729,3 +730,146 @@ def test_request_suspicious_request_block_match_response_headers(client, test_sp
+        test_user = User.objects.create(username="fred2")
+        test_user.set_password("secret")
+        test_user.save()
+        assert not get_user(client).is_authenticated
+        client.login(username="fred2", password="secret")
+        assert get_user(client).is_authenticated
+        login_span = test_spans.find_span(name="django.contrib.auth.login")
+        assert login_span
+        assert login_span.get_tag(user.ID) == "1"
+        assert login_span.get_tag(APPSEC.USER_LOGIN_EVENT_PREFIX + ".success.track") == "true"
+        assert login_span.get_tag(APPSEC.USER_LOGIN_EVENT_PREFIX + ".success.auto.mode") == "safe"
+
+
+@pytest.mark.django_db
+def test_django_login_sucess_safe_is_default_if_wrong(client, test_spans, tracer):
+    from django.contrib.auth import get_user
+    from django.contrib.auth.models import User
+
+    with override_global_config(dict(_appsec_enabled=True, _automatic_login_events_mode="foobar")):
+        test_user = User.objects.create(username="fred")
+        test_user.set_password("secret")
+        test_user.save()
+        client.login(username="fred", password="secret")
+        assert get_user(client).is_authenticated
+        login_span = test_spans.find_span(name="django.contrib.auth.login")
+        assert login_span.get_tag(user.ID) == "1"
+
+
+@pytest.mark.django_db
+def test_django_login_sucess_safe_is_default_if_missing(client, test_spans, tracer):
+    from django.contrib.auth import get_user
+    from django.contrib.auth.models import User
+
+    with override_global_config(dict(_appsec_enabled=True)):
+        test_user = User.objects.create(username="fred")
+        test_user.set_password("secret")
+        test_user.save()
+        client.login(username="fred", password="secret")
+        assert get_user(client).is_authenticated
+        login_span = test_spans.find_span(name="django.contrib.auth.login")
+        assert login_span.get_tag(user.ID) == "1"
+
+
+@pytest.mark.django_db
+def test_django_login_failure_user_doesnt_exists(client, test_spans, tracer):
+    from django.contrib.auth import get_user
+
+    with override_global_config(dict(_appsec_enabled=True, _automatic_login_events_mode="extended")):
+        assert not get_user(client).is_authenticated
+        client.login(username="missing", password="secret2")
+        assert not get_user(client).is_authenticated
+        login_span = test_spans.find_span(name="django.contrib.auth.login")
+        assert login_span.get_tag(APPSEC.USER_LOGIN_EVENT_PREFIX + ".failure.track") == "true"
+        assert login_span.get_tag(APPSEC.USER_LOGIN_EVENT_PREFIX + ".failure." + user.ID) == "missing"
+        assert login_span.get_tag(APPSEC.USER_LOGIN_EVENT_PREFIX + ".failure." + user.EXISTS) == "false"
+        assert login_span.get_tag(APPSEC.USER_LOGIN_EVENT_PREFIX + ".failure.auto.mode") == "extended"
+
+
+@pytest.mark.django_db
+def test_django_login_sucess_safe_but_user_set_login(client, test_spans, tracer):
```

Comment:
Ah this demonstrates some interesting behavior, which I think I noted in the code earlier. Maybe we should document this somewhere for users?

Reply:
Yup. Will note this also on the new change on the `configuration.rst`.

Reply:
Done